taginfo.cpp

```diff
@@ -145,6 +153,13 @@ int read_style_file( const std::string &filename, export_list *exlist )
     temp.type.assign(datatype);
     temp.flags = parse_tag_flags(flags, lineno);
 
+    try {
+        temp.flags |= tagtypes.at(temp.type);
+    } catch (std::out_of_range const &e) {
+        fprintf(stderr, "Bad datatype '%s' in style file line %d\n", datatype, lineno);
```

Comment:
An unknown datatype should be a warning, not an error. e.g. `character varying`, `varchar`, `char(255)` are all valid types for arbitrary OSM data.

Also, it shouldn't warn if using Lua transforms because any column type can be valid there, including stranger ones like enumerated types, `text[]`, `hstore`, and `jsonb`